### PR TITLE
Added MapMiddleware extensions in EndpointBuilderExtension.

### PR DESCRIPTION
**Pull request by pavinan**
Summary of the changes 
 - Added MapMiddleware extensions that are similar to route builder extensions.

Addresses #14514

### CODE REVIEW

File: src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs [hunk 1]
```diff
@@ -187,5 +187,108 @@ public static class EndpointRouteBuilderExtensions
...
+
+        /// <summary>
+        /// Adds a <see cref="RouteEndpoint"/> to the <see cref="IEndpointRouteBuilder"/> for the specified <paramref name="pattern"/>, and
+        /// <paramref name="action"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IEndpointRouteBuilder"/>.</param>
+        /// <param name="pattern">The route pattern.</param>
+        /// <param name="action">The action to apply to the <see cref="IApplicationBuilder"/>.</param>
+        /// <returns>A <see cref="IEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+        public static IEndpointConventionBuilder MapMiddleware(
```
Comment (Tratcher) [inline src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs]: Middleware is not the best term here. How well would these work if they were just a overloads of Map? Would the arguments conflict too much?

Comment (pavinan) [reply src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs]: How about MapMiddlewareEndpoint? It will be similar to MapMiddlewareRoute.

**Review by rynowak — CHANGES_REQUESTED** [review]
@pavinan - please open an issue and start a discussion about this. We need to think about whether we want to expose this functionality in this, and what set of methods would be required to do that.

**Comment by anurse** [comment]
I'm going to close this PR until we finish some discussion about design in an issue. We can always re-open this or accept a new PR once a design is finalized.

**Pull request closed without merging**